### PR TITLE
[BACKEND] Price Change Notifications

### PR DESCRIPTION
**Pull request by umutoksuz**
## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Description

Below mentioned tasks are done to implement notification system:
- A function is created to populate a notification when price change event occurs
- Tests for notification system 
 
Additionally, delete product functionality is added to vendor/product

# Test results

```
System check identified no issues (0 silenced).
....................................................................
----------------------------------------------------------------------
Ran 68 tests in 5.453s

OK
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by furkannane — APPROVED** [review]
Changes are appropriate.

File: ecommerce/backend/API/tests/test_notification.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            "long_description": "anoacnancacnoewcnwocnoınvroeı3vnoıwvnroeıvnıoernvoernvoernvoıenvoıernvoerınvıernvoıernvoıernvroeınvoıernvorıenvoeırnvoıernvoıernvenvoernvoıernverı",
+            "discount": 0.18,
+            "brand_id": 1,
+            "subcategory_id": 11,
+            "images": [],
+            "image_urls_delete": []
+        }
+        response = self.client.put(reverse('vendor_product'), body, 'json')
+        notification = Notification.objects.filter(user=self.c_user).first()
+        self.assertNotEqual(notification, None)
```
Comment (omerfdeniz) [inline ecommerce/backend/API/tests/test_notification.py]: Please check the status code and response in your tests. Also do not forget to adhere to our response convention in manage_notifications()
```suggestion
        self.assertNotEqual(notification, None)
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.data["status"]["successful"], True)
```

Comment (omerfdeniz) [reply ecommerce/backend/API/tests/test_notification.py]: Please also test having no notification when a PUT request with the same price is sended

**Review by omerfdeniz — CHANGES_REQUESTED** [review]
Please check the response add the test provided in the comment

**Review by bekir96 — APPROVED** [review]
All test results looks good and migration handling is correct, you can merge

**Review by omerfdeniz — APPROVED** [review]
Great work, now you can merge

**Pull request merged**